### PR TITLE
Updated all global variables to use Request class

### PR DESCRIPTION
**Pull request by cezarlz**

### CODE REVIEW

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php [hunk 1]
```diff
@@ -177,15 +177,16 @@ public function checkout_assets()
...
 	{
-		if (empty($_POST['ebanx_token']) ||
-			empty($_POST['ebanx_masked_card_number']) ||
-			empty($_POST['ebanx_brand']) ||
-			empty($_POST['ebanx_billing_cvv'])
+		
+		if (!WC_EBANX_Request::has('ebanx_token') ||
+			!WC_EBANX_Request::has('ebanx_masked_card_number') ||
+			!WC_EBANX_Request::has('ebanx_brand') ||
+			!WC_EBANX_Request::has('ebanx_billing_cvv')
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php]: `empty($_POST[...])` is not the same as `WC_EBANX_Request::has(...)`, this should be replaced with `empty(WC_EBANX_Request::read(..., null))`

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php [hunk 2]
```diff
@@ -177,15 +177,16 @@ public function checkout_assets()
...
 		) {
 			throw new Exception('MISSING-CARD-PARAMS');
 		}
 
-		if (empty($_POST['ebanx_is_one_click']) && empty($_POST['ebanx_device_fingerprint'])) {
+		if (!WC_EBANX_Request::has('ebanx_is_one_click') && !WC_EBANX_Request::has('ebanx_device_fingerprint')) {
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php]: Same as above

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php [hunk 3]
```diff
@@ -194,19 +195,19 @@ protected function request_data($order)
...
-			if ($this->configs->settings['credit_card_instalments'] > 1 && isset($_POST['ebanx_billing_instalments'])) {
-				$data['payment']['instalments'] = $_POST['ebanx_billing_instalments'];
+			if ($this->configs->settings['credit_card_instalments'] > 1 && WC_EBANX_Request::has('ebanx_billing_instalments')) {
+				$data['payment']['instalments'] = WC_EBANX_Request::read('ebanx_billing_instalments');
 			}
 		}
 
-		if (!empty($_POST['ebanx_device_fingerprint'])) {
-			$data['device_id'] = $_POST['ebanx_device_fingerprint'];
+		if (WC_EBANX_Request::has('ebanx_device_fingerprint')) {
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php]: Same as above

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php (hunk 1 above)
Comment (SparK-Cruz) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php]: `empty($_POST)` should be replaced by `empty(WC_EBANX_Request::read($key, null))`

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-debit-card-gateway.php [hunk 4]
```diff
@@ -87,8 +87,8 @@ public function payment_fields()
 	 */
 	protected function request_data($order)
 	{
-		if ( empty(WC_EBANX_Request::read('ebanx_debit_token'))
-			|| empty(WC_EBANX_Request::read('ebanx_billing_cvv')) ) {
+		if ( empty(WC_EBANX_Request::read('ebanx_debit_token', null))
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-debit-card-gateway.php]: Nice!

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php (hunk 2 above)
Comment (SparK-Cruz) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php]: same as above

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-gateway.php [hunk 5]
```diff
@@ -491,15 +491,13 @@ protected function request_data($order)
 					return array(
 						'name' => $product['name'],
 						'unit_price' => $product['line_subtotal'],
-					  'quantity' => $product['qty'],
+					  	'quantity' => $product['qty'],
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-gateway.php]: Great!

File: woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php (hunk 3 above)
Comment (SparK-Cruz) [inline woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php]: same as above

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 6]
```diff
@@ -306,7 +306,7 @@ public function print_button() {
 				break;
 			default:
 				$messages = array(
-					'instalments' => 'Number of installments',
+					'instalments' => 'Number of instalments',
```
Comment (gpressutto5) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: Sweet!

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 7]
```diff
@@ -244,28 +244,28 @@ protected function customer_has_ebanx_required_data() {
...
+		WC_EBANX_Request::set($names['ebanx_billing_brazil_birth_date'], get_user_meta( $this->userId, '_ebanx_billing_brazil_birth_date', true ));
+
+		WC_EBANX_Request::set('billing_postcode', $this->get_user_billing_address()['postcode']);
+		WC_EBANX_Request::set('billing_address_1', $this->get_user_billing_address()['address_1']);
+		WC_EBANX_Request::set('billing_city', $this->get_user_billing_address()['city']);
+		WC_EBANX_Request::set('billing_state', $this->get_user_billing_address()['state']);
+
+		return WC_EBANX_Request::has('ebanx-one-click-token')
+			&& WC_EBANX_Request::has('ebanx-credit-card-installments')
+			&& WC_EBANX_Request::has('ebanx-one-click-cvv')
```
Comment (SparK-Cruz) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: `ebanx-one-click-token`, `ebanx-credit-card-installments`, `ebanx-one-click-cvv`: same, check empty.
The if should be all "||" and the has lines negated.

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 8]
```diff
@@ -244,28 +244,28 @@ protected function customer_has_ebanx_required_data() {
...
+			&& WC_EBANX_Request::has($names['ebanx_billing_brazil_document'])
+			&& WC_EBANX_Request::has($names['ebanx_billing_brazil_birth_date'])
```
Comment (SparK-Cruz) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: negate these two, change if to ||

File: woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php [hunk 9]
```diff
@@ -244,28 +244,28 @@ protected function customer_has_ebanx_required_data() {
...
+			&& WC_EBANX_Request::has($names['ebanx_billing_brazil_document'])
+			&& WC_EBANX_Request::has($names['ebanx_billing_brazil_birth_date']);
```
Comment (SparK-Cruz) [inline woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php]: same as others, check empty with null

**Pull request merged**